### PR TITLE
add nano g1

### PR DESCRIPTION
**Pull request by neilhao**
Meshtastic Mesh Device Nano Edition G1 has been added. The code could be compiled successfully. 
This is my first commit to Meshtastic, hope I didn't get anything wrong.

The hardware Info:
https://uniteng.com/wiki/doku.php?id=meshtastic:nano

Neil

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -19,6 +19,7 @@ default_envs = tbeam
 ;default_envs = rak4631_5005
 ;default_envs = rak4631_5005_eink
 ;default_envs = rak4631_19003
+default_envs = nano-g1
 ;default_envs = meshtastic-diy-v1
```
Comment (caveman99) [inline platformio.ini:23]: Please don't edit the master platformio.ini unless absolutely necessary. This change can be reverted without harm.

File: src/configuration.h [hunk 2]
```diff
@@ -134,7 +134,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 // The SH1106 controller is almost, but not quite, the same as SSD1306
 // Define this if you know you have that controller or your "SSD1306" misbehaves.
-//#define USE_SH1106
+#define USE_SH1106
```
Comment (caveman99) [inline src/configuration.h:138]: Please revert that change, the type of controller (1106/1306) is auto-detected now if it is present on the i2c bus. see https://github.com/meshtastic/Meshtastic-device/blob/a8ef3967cde22993608c7adba051b3cea502ba30/src/debug/i2cScan.h#L6-L25

**Review by caveman99 — CHANGES_REQUESTED** [review]
Pretty good for a firest PR. Please review my comments, other than that it's good to go.

**Review by caveman99 — APPROVED** [review]

**Comment by caveman99** [comment]
Caveat: If this is approved, the protobufs in master need to be updated before it is merged.

**Review by sachaw — CHANGES_REQUESTED** [review]
Looks great, just some line cleanup, then we can squash and merge.

**Comment by mc-hamster** [comment]
This really is beautiful. Way to go!

**Pull request merged**